cdap-ui/app/cdap/components/AbstractWidget/HierarchyWidget/HierarchyTreeNode/index.tsx

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as React from 'react';
+import IconButton from '@material-ui/core/IconButton';
+import { makeStyles, Theme, withStyles } from '@material-ui/core/styles';
+import Box from '@material-ui/core/Box';
+import { Paper, TextField } from '@material-ui/core';
+import DeleteIcon from '@material-ui/icons/Delete';
+import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
+import KeyboardArrowRightIcon from '@material-ui/icons/KeyboardArrowRight';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import { toJsonHandler, handleChildren, flatToTree, removeFamily } from '../dataHandler';
+import { IHierarchyProps } from 'components/AbstractWidget/HierarchyWidget';
+import HierarchyTreeLeaf from 'components/AbstractWidget/HierarchyWidget/HierarchyTreeNode/HierarchyTreeLeaf';
+import HierarchyPopoverButton from 'components/AbstractWidget/HierarchyWidget/HierarchyTreeNode/PopoverButton';
+import InputFieldWrapper from 'components/AbstractWidget/HierarchyWidget/HierarchyTreeNode/InputFieldWrapper';
+import HierarchyTree from 'components/AbstractWidget/HierarchyWidget/HierarchyTree';
+import If from 'components/If';
+import uuidV4 from 'uuid/v4';
+
+const RowInputContainer = withStyles(() => {
+  return {
+    root: {
+      padding: '2px 10px 2px 0px',
+      display: 'grid',
+      marginTop: '2px',
+      gridTemplateColumns: '4fr 2fr 0fr',
+      alignItems: 'center',
+      gridTemplateRows: '28px',
+      position: 'relative',
+      borderRadius: '3px',
+    },
+  };
+})(Paper);
+
+const AutocompleteContainer = withStyles(() => {
+  return {
+    root: {
+      padding: '2px 10px 2px 0px',
+      marginTop: '2px',
+      position: 'relative',
+      borderRadius: '3px',
+    },
+  };
+})(Paper);
+
+const RowButtonsWraper = withStyles(() => {
+  return {
+    root: {
+      display: 'grid',
+      gridTemplateColumns: '1fr 1fr',
+      justifyItems: 'end',
+    },
+  };
+})(Box);
+
+const ButtonIcon = withStyles(() => {
+  return {
+    root: {
+      padding: '6.5px',
+    },
+  };
+})(IconButton);
+
+const HiddenBtn = withStyles(() => {
```

Comment:
Can these two be combined based on a property? Or have `makeStyle` accept a hidden property and return a class name that can be used on `IconButton` component